File: one-to-one-videocall/src/main/java/fr/marcworld/webrtcdemo/onetoonevideocall/dtos/CallUserServerEvent.java
package fr.marcworld.webrtcdemo.onetoonevideocall.dtos;

public class CallUserServerEvent extends UserServerEvent {

    private int callerUserId;
    private int otherUserId;

    public CallUserServerEvent() {
        super();
    }

    public CallUserServerEvent(UserServerEventCode code, int callerUserId, int otherUserId) {
        super(code);
        this.callerUserId = callerUserId;
        this.otherUserId = otherUserId;
    }

    public int getCallerUserId() {
        return callerUserId;
    }

    public void setCallerUserId(int callerUserId) {
        this.callerUserId = callerUserId;
    }

    public int getOtherUserId() {
        return otherUserId;
    }

    public void setOtherUserId(int otherUserId) {
        this.otherUserId = otherUserId;
    }
}

File: one-to-one-videocall/src/main/resources/static/scripts/model/CallUserServerEvent.js
import UserServerEventCode from './UserServerEventCode.js';
import UserServerEvent from './UserServerEvent.js';

class CallUserServerEvent extends UserServerEvent {

    /**
     * @param {{code: UserServerEventCode, callerUserId: Number, otherUserId: Number}} properties
     */
    static fromProperties(properties) {
        return new CallUserServerEvent(properties.code, properties.callerUserId, properties.otherUserId);
    }

    /**
     * @param {UserServerEventCode} code
     * @param {Number} callerUserId
     * @param {Number} otherUserId
     */
    constructor(code, callerUserId, otherUserId) {
        super(code);
        this._callerUserId = callerUserId;
        this._otherUserId = otherUserId;
    }

    get callerUserId() {
        return this._callerUserId;
    }

    get otherUserId() {
        return this._otherUserId;
    }
}

export default CallUserServerEvent;